WordPress/Classes/ViewRelated/Blog/Site Settings/DiscussionSettingsViewController.swift

```diff
@@ -268,6 +277,7 @@ open class DiscussionSettingsViewController: UITableViewController {
             }
 
             self?.settings.commentsThreading = newThreadingDepth
+            WPAnalytics.trackSettingsChange("site_settings_discussion", fieldName: "comments_threading", value: selected as Any)
```

Comment:
I think most of these settings keys should be stored as constants somewhere (and likely field names), as the same literals are reproduced in many places and a typo could mean we end up tracking the wrong thing. This comment applies to basically every `trackSettingsChange` call 🙂

Reply:
I can see this point, but all of them are stored as private constants in [BlogServiceRemoteREST](https://github.com/wordpress-mobile/WordPressKit-iOS/blob/c5e11d569989e6782330200121545832c2a3b3d5/WordPressKit/BlogServiceRemoteREST.m) and it feels like overkill to expose these constants, or duplicate them in a public fashion.

Reply:
Hmm, I guess I mean if nothing else we could have a constant in `DiscussionSettingsViewController` for `let tracksDiscussionSettingsKey = "site_settings_discussion"` and so forth?

Reply:
I refactored the tracking a bit more on that view, added a constant, and moved all of the tracking to a helper func.